net: lib: openthread: Handle OT transmission in Thread task

MarekPorwisz opened the pull request:
OpenThread API is not thread-safe.
Moved creation of otMessage to the Thread task and created api
for passing it properly.
This way it should be less possible for an issue to occure eg.
during message buffer allocation.

Signed-off-by: Marek Porwisz <marek.porwisz@nordicsemi.no>

MarekPorwisz committed:
net: lib: openthread: Handle OT transmission in Thread task

OpenThread API is not thread-safe.
Moved creation of otMessage to the Thread task and created api
for passing it properly.
This way it should be less possible for an issue to occure eg.
during message buffer allocation.

Signed-off-by: Marek Porwisz <marek.porwisz@nordicsemi.no>

rlubos commented on subsys/net/l2/openthread/openthread.c:
```diff
@@ -299,42 +299,17 @@ static enum net_verdict openthread_recv(struct net_if *iface,
-	otMessage *message;
-	otMessageSettings settings;
 
 	NET_DBG("Sending Ip6 packet to ot stack");
```
This debug is also present in `openthread_handle_frame_to_send` now, IMO it's enough to have it in one place.

rlubos commented on subsys/net/lib/openthread/platform/radio.c:
```diff
@@ -302,6 +306,41 @@ static void openthread_handle_received_frame(otInstance *instance,
+	for (buf = pkt->buffer; buf; buf = buf->frags) {
+		if (otMessageAppend(message, buf->data,
+				    buf->len) != OT_ERROR_NONE) {
+
```
An unneeded empty line.

rlubos commented on subsys/net/lib/openthread/platform/radio.c:
```diff
@@ -302,6 +306,41 @@ static void openthread_handle_received_frame(otInstance *instance,
+	}
+
+	if (otIp6Send(instance, message) != OT_ERROR_NONE) {
+		NET_ERR("Error while calling otIp6Send");
```
I see it was like this before, but I wonder - will `otIp6Send` free the message in case of error? We don't call `otMessageFree(message)` here, could you verify that?

MarekPorwisz replied:
Yes, otIp6Send calls free if it fails to send the message.

MarekPorwisz committed:
tests: openthread: Test passing net_pkt to OpenThread thread

Passing TX messages from sockets was reworked for openthread
thus tests need to be created/updated in order to pass.

Signed-off-by: Marek Porwisz <marek.porwisz@nordicsemi.no>

rlubos reviewed (approved):
LGTM

jukkar commented on subsys/net/l2/openthread/openthread.c:
```diff
@@ -299,42 +299,15 @@ static enum net_verdict openthread_recv(struct net_if *iface,
 
```
In commit subject, it is enough to say `net: openthread: ....`, the lib is redundant information here.

jukkar reviewed (approved):
LGTM
